packages/codd/scripts/codd.py

```
@@ -343,6 +343,182 @@ def cmd_validate(root: Path, config: cc.CoddConfig) -> int:
+
+
+def compute_impact_result(root: Path, config: cc.CoddConfig, ref: str) -> ImpactResult:
+    """scan → diff 突合 → 影響分析までを行い ImpactResult を返す。"""
+    _warn_if_not_git_root(root)
+    result = scan_project(root, config)
+    changed_paths, deleted_paths = diff_changed_paths(root, ref)
+
+    path_to_id = {node.path: node.node_id for node in result.nodes}
+    changed_ids = {path_to_id[p] for p in changed_paths if p in path_to_id}
+
+    # 削除された scope 内ドキュメント（下流が dangling 化する可能性）。
```

Ruff (0.15.18)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)